app/actions/user.ts

```diff
@@ -24,62 +24,50 @@ export const loadAccountDetails = (params: {
     try {
-      const bctContract = new ethers.Contract(
-        addresses["mainnet"].bct,
-        IERC20.abi,
-        params.provider
-      );
-      const nctContract = new ethers.Contract(
-        addresses["mainnet"].nct,
-        IERC20.abi,
-        params.provider
-      );
-      const mco2Contract = new ethers.Contract(
-        addresses["mainnet"].mco2,
-        IERC20.abi,
-        params.provider
-      );
-      const uboContract = new ethers.Contract(
-        addresses["mainnet"].ubo,
-        IERC20.abi,
-        params.provider
-      );
-      const nboContract = new ethers.Contract(
-        addresses["mainnet"].nbo,
-        IERC20.abi,
-        params.provider
-      );
-      const usdcContract = new ethers.Contract(
-        addresses["mainnet"].usdc,
-        IERC20.abi,
-        params.provider
-      );
-      const klimaContract = new ethers.Contract(
-        addresses["mainnet"].klima,
-        IERC20.abi,
-        params.provider
-      );
-      const sklimaContract = new ethers.Contract(
-        addresses["mainnet"].sklima,
-        IERC20.abi,
-        params.provider
-      );
-      const wsklimaContract = new ethers.Contract(
-        addresses["mainnet"].wsklima,
-        wsKlima.abi,
-        params.provider
-      );
-      const pKlimaContract = new ethers.Contract(
-        addresses["mainnet"].pklima,
-        IERC20.abi,
-        params.provider
-      );
-      // remember to add this to addresses object
-      const klimaDomainContract = new ethers.Contract(
-        addresses["mainnet"].klimaNameService,
-        PunkTLD.abi,
-        params.provider
-      );
+      const bctContract = getContractByToken({
+        token: "bct",
+        provider: params.provider,
+      });
+      const nctContract = getContractByToken({
+        token: "nct",
+        provider: params.provider,
+      });
+      const mco2Contract = getContractByToken({
+        token: "mco2",
+        provider: params.provider,
+      });
+      const uboContract = getContractByToken({
+        token: "ubo",
+        provider: params.provider,
+      });
+      const nboContract = getContractByToken({
+        token: "nbo",
+        provider: params.provider,
+      });
+      const usdcContract = getContractByToken({
+        token: "usdc",
+        provider: params.provider,
+      });
+      const klimaContract = getContractByToken({
+        token: "klima",
+        provider: params.provider,
+      });
+      const sklimaContract = getContractByToken({
+        token: "sklima",
+        provider: params.provider,
+      });
+      const wsklimaContract = getContractByToken({
+        token: "wsklima",
+        provider: params.provider,
+      });
+      const pKlimaContract = getContractByToken({
+        token: "pklima",
+        provider: params.provider,
+      });
+      const klimaDomainContract = getContractByToken({
+        token: "klimaNameService",
+        provider: params.provider,
+      });
```

Comment:
this could be loopified to save a few lines
```ts
const [bctContract, nctContract, ...etc] = ["bct", "nct", ...etc].map(token => getContractByToken({ token, provider}))
```

Reply:
This is the refactoring I want to do in a follow-up

```TS
const contracts = createContractsObject("bct", "nct", "ubo"); 
// => {
    bct: ethers.Contract,
    nct: ethers.Contract
    ubo: ethers.contract
...
}